openshift/catalog-sync/pkg/check/image_check.go

```diff
@@ -0,0 +1,79 @@
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"oras.land/oras-go/v2"
+	"os"
+	"path/filepath"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type Checks struct {
+	ImageChecks      []ImageCheck
+	FilesystemChecks []FilesystemCheck
+	CatalogChecks    []CatalogCheck
+}
+
+type ImageCheckFunc func(ctx context.Context, root ocispec.Descriptor, target oras.ReadOnlyTarget) error
+type FilesystemCheckFunc func(ctx context.Context, imageFS fs.FS) error
+type CatalogCheckFunc func(ctx context.Context, fbcFS fs.FS) error
+
+type ImageCheck struct {
+	Name string
+	Fn   ImageCheckFunc
+}
+
+type FilesystemCheck struct {
+	Name string
+	Fn   FilesystemCheckFunc
+}
+
+type CatalogCheck struct {
+	Name string
+	Fn   CatalogCheckFunc
+}
+
+type Error struct {
+	CheckName string
+	Err       error
+}
+
+func (e Error) Error() string {
+	return fmt.Sprintf("%s: %s", e.CheckName, e.Err)
+}
+
+func Check(ctx context.Context, store oras.ReadOnlyTarget, ociDir string, checks Checks) error {
+	desc, err := store.Resolve(ctx, "v1")
+	if err != nil {
+		return fmt.Errorf("resolve descriptor: %w", err)
+	}
+
+	unpackPath := filepath.Join(ociDir, "fs")
+	if _, err := os.Stat(unpackPath); err != nil {
+		return fmt.Errorf("expected extracted filesystem at %q: %w", unpackPath, err)
+	}
```

Comment:
Move this inside a check for `len(checks.FilesystemChecks) > 0` (and put the check running loop inside that conditional bloack as well)?